lib/Models/BoxDrawing.ts

```diff
@@ -533,7 +543,11 @@ export default class BoxDrawing {
     const handlePick = (click: MouseClick) => {
       const pick = scene.pick(click.position);
       const entity = pick?.id;
-      if (entity === undefined || !isInteractable(entity)) {
+      if (
+        entity === undefined ||
+        !isInteractable(entity) ||
+        !this.dataSource.entities.contains(entity)
```

Comment:
Without this extra check, the `handlePick` function is run even when the picked entity does not belong to this box. As a result, when multiple BoxDrawings are active on the map, we capture the map state multiple times and restore it back multiple times resulting in breaking of the panning and zooming of the map:

https://github.com/TerriaJS/terriajs/blob/b6502f7a70c76a34980833734396dad381ac826e/lib/Models/BoxDrawing.ts#L563-L566